File: 08/input_parser.rb
require_relative "routing"

module InputParser
  def self.parse(input)
    navigation_line = input.shift

    routing = Routing.new(navigation_line)

    input.each do |node_line|
      parse_node(node_line, routing)
    end

    routing
  end

  def self.parse_node(node_line, routing)
    node, going_to = node_line.split(" = ")
    left, right = going_to.scan(/[A-Z]+/)

    routing.add_node(node, [left, right])
  end
end

File: 09/input_parser.exs
defmodule InputParser do
  # e.g.
  # 0 3 6 9 12 15
  # 1 3 6 10 15 21
  # 10 13 16 21 30 45
  #
  def parse(input) do
    input
    |> Enum.map(fn line ->
      String.split(line, " ", trim: true)
      |> Enum.map(&String.to_integer/1)
    end)
  end
end
